dateutil/parser/_parser.py

```diff
@@ -1313,7 +1315,8 @@ def __init__(self):
 
     def parse(self, tzstr):
         res = self._result()
-        l = _timelex.split(tzstr)
+        l = [x for x in re.split(r'([,:.]|[a-zA-Z]+|[0-9]+)',tzstr) if x]
```

Comment:
If we see something like `"foo,:"bar` with two of `[,:.]` in a row is that indicative of an invalid string?

Reply:
The parser will fail in that case. I will push for the tokenizer part not doing anything else that tokenize and just let the parser naturaly fail to parse a consecutive `[,:.]`.

Reply:
The reason why I ask is to see if the `if x` clause is necessary.

Reply:
This is necessary for Python<3.6. This is due to the way `re.split` works. I have made a failing commit to prove this in b15f38a. [Here](https://travis-ci.org/dateutil/dateutil/builds/316445999?utm_source=github_status&utm_medium=notification) are the CI results of that commit:

Reply:
I'm not sure I understand this regular expression completely, but I think the only problem with this approach that I see is that any `split` based function is going to discard the tokens themselves. Are there really no constraints on what the inputs can be for valid TZ strings?

Also, with the `if x` filter, how would you distinguish between `One,Two` and `One,,Two`?

```python
In [6]: [x for x in re.split(r'[.,;]', 'One,Two') if x]
Out[6]: ['One', 'Two']

In [7]: [x for x in re.split(r'[.,;]', 'One,;Two') if x]
Out[7]: ['One', 'Two']
```

Reply:
You forgot the capturing group:

```python
>>> [x for x in re.split(r'([,:.]|[a-zA-Z]+|[0-9]+)','One,Two') if x]
['One', ',', 'Two']
>>> [x for x in re.split(r'([,:.]|[a-zA-Z]+|[0-9]+)','One,;Two') if x]
['One', ',', ';', 'Two']
```

Reply:
Oh wow, that's a neat trick. I didn't realized it worked like that. Good to know.